test/regression/generateRegressionTests.ts

```diff
@@ -19,47 +19,124 @@ while (matched = componentExtractor.exec(taggedForRegression)) {
     }
 }
 
+const LOCAL_DEV_ID = `local-dev-test-${new Date().toISOString()}`;
 const regressionTest = async (t: TestContext, config: IConfig, component: string) => {
+    if (process.env.CI) {
+        config.serverUrl = `http://${process.env.SAUCE_USERNAME}:${process.env.SAUCE_ACCESS_KEY}@ondemand.saucelabs.com:80/wd/hub`;
+        config.sauceLabs.name = component;
+        config.sauceLabs.build = process.env.TRAVIS_JOB_NUMBER || LOCAL_DEV_ID;
+        config.sauceLabs.tunnelIdentifier = process.env.TRAVIS_JOB_NUMBER;
+    }
+
     const snappit = new Snappit(config);
     const driver = await snappit.start();
     await util.go(driver, component);
 
-    for (const e of await driver.findElements(By.css(util.selectors.visreg))) {
-        const sectionName = await e.getAttribute("data-visreg");
-        t.log(`  ${sectionName}:`);
-        await util.snapshot(t, e);
-        t.log("    ✔ DOM Snapshot");
-        await driver.executeScript("window.scroll(0, 0);");
-        await snappit.snap(`{browserName}/${sectionName}`, e as WebElement);
-        t.log("    ✔ Image Snapshot");
-    }
+    try {
+        for (const e of await driver.findElements(By.css(util.selectors.visreg))) {
+            const sectionName = await e.getAttribute("data-visreg");
+            t.log(`  ${sectionName}:`);
+            await util.snapshot(t, e);
+            t.log("    ✔ DOM Snapshot");
+
+            if (!process.env.CI) {
+                // this is really not worth it for saucelabs, you can just look at the run results anyway
+                await snappit.snap(`{browserName}/${sectionName}`, e as WebElement);
+                t.log("    ✔ Image Snapshot");
+            }
+        }
 
-    await snappit.stop();
+        process.env.CI && await snappit.setSauceLabsJobResult(true);
+    } catch (e) {
+        process.env.CI && await snappit.setSauceLabsJobResult(false);
```

Comment:
Don't use process.env.CI use another var that represents that the user is running saucelabs tests.

```js
const runningSauce = process.env.SAUCE_USERNAME && process.env.SAUCE_ACCESS_KEY;
```

Or something.